lib/awful/widget/inputbox.lua

```diff
@@ -398,21 +374,62 @@ function inputbox:run(callbacks)
         end
 
         -- Update cycle
-        self:update(self._private.text, cursor_pos)
+        self.widget:update {
+            text = self:get_text(),
+            cursor_pos = cursor_pos,
+            self = self
+        }
 
-        if callbacks.pressed then
-            callbacks.pressed(key, mod_keys)
-        end
+        self:emit_signal("key_pressed", mod_keys, key)
     end)
 end
 
---- Creates a new inpubox widget
--- @tparam table args Arguments for the inpubox widget
--- @tparam string args.text The text to display in the inpubox
+local default_widget = {
+    template = {
+        {
+            {
+                {
+                    widget = wibox.widget.textbox,
+                    text = "",
+                    halign = "left",
+                    valign = "center",
+                    id = "inputtext",
+                },
+                widget = wibox.container.margin,
+                margins = 5,
+            },
+            widget = wibox.container.constraint,
+            strategy = "exact",
+            width = 400,
+            height = 50,
+        },
+        widget = wibox.container.background,
+        bg = "#212121",
+        fg = "#F0F0F0",
+        border_color = "#414141",
+        border_width = 2,
+        shape = gshape.rounded_rect,
+        forced_width = 300,
+        forced_height = 50,
+    },
```

Comment:
No, I just thought that if the user doesn't provide its own template that it would make sense to provide a widget that you can see easier. (No borders but the same background could make the user think that there is nothing)

Reply:
But the other side of the coin is that it now becomes harder to embed the widget into any custom widget tree, because the user needs to undo your opinionated default styling.
E.g. we don't usually force a rounded shape, or restrict widget sizes by default.

The general usage of the widget system is "when you want to add/change something, wrap it". And in the general Unix philosophy "do one thing well", each widget only does the specific things it is designed/intended for.

Maybe another point to hear opinions for, though.